src/videopython/base/transforms.py

```diff
@@ -89,3 +90,41 @@ def apply(self, video: Video) -> Video:
             )
         video.frames = np.array(frames_copy)
         return video
+
+
+class ResampleFPS(Transformation):
+    def __init__(self, new_fps: int | float):
+        self.new_fps = float(new_fps)
+
+    def _downsample(self, video: Video) -> Video:
+        target_frame_count = int(len(video.frames) * (self.new_fps / video.fps))
+        new_frame_indices = np.round(np.linspace(0, len(video.frames) - 1, target_frame_count)).astype(int)
+        video.frames = video.frames[new_frame_indices]
+        video.fps = self.new_fps
+        return video
```

Comment:
For the future I would consider using interpolation also for downsampling. We can also think about more advanced techniques like ones using optical flow.